src/javascript/SelectorTypes/Picker/PickerDialog/SelectionHandler.jsx

```diff
@@ -0,0 +1,166 @@
+import React from 'react';
+import {shallowEqual, useDispatch, useSelector} from 'react-redux';
+import {useNodeInfo} from '@jahia/data-helper';
+import {useContentEditorConfigContext} from '~/contexts';
+import {useQuery} from '@apollo/react-hooks';
+import {GET_PICKER_NODE} from '~/SelectorTypes/Picker';
+import {useEffect, useRef} from 'react';
+import {
+    cePickerContextSite,
+    cePickerMode,
+    cePickerOpenPaths,
+    cePickerPath,
+    cePickerSetSelection,
+    cePickerSite
+} from '~/SelectorTypes/Picker/Picker2.redux';
+import {encodeJCRPath} from '~/utils';
+import {registry} from '@jahia/ui-extender';
+import {getItemTarget} from '~/SelectorTypes/Picker/accordionItems/accordionItems';
+import {getAccordionMode, getDetailedPathArray, getPathWithoutFile} from '~/SelectorTypes/Picker/Picker2.utils';
+import {batchActions} from 'redux-batched-actions';
+import PropTypes from 'prop-types';
+import {configPropType} from '~/SelectorTypes/Picker/configs/configPropType';
+import {LoaderOverlay} from '~/DesignSystem/LoaderOverlay';
+
+function getSite(selectedItem) {
+    const pathElements = selectedItem.split('/');
+    return (pathElements[1] === 'sites') ? pathElements[2] : 'systemsite';
+}
+
+export const SelectionHandler = ({initialSelectedItem, editorContext, pickerConfig, children}) => {
+    const state = useSelector(state => ({
+        mode: state.contenteditor.picker.mode,
+        path: state.contenteditor.picker.path,
+        openPaths: state.contenteditor.picker.openPaths,
+        site: state.contenteditor.picker.site,
+        contextSite: state.contenteditor.picker.contextSite,
+        jcontentMode: state.jcontent.mode,
+        jcontentPath: state.jcontent.path
+    }), shallowEqual);
+
+    const dispatch = useDispatch();
+
+    const currentFolderInfo = useNodeInfo({path: state.path}, {skip: !state.path});
+    console.log(state.path, currentFolderInfo.loading);
+    const {lang, uilang} = useContentEditorConfigContext();
+
+    const paths = (Array.isArray(initialSelectedItem) ? initialSelectedItem : [initialSelectedItem]).filter(f => f);
+    const nodesInfo = useQuery(GET_PICKER_NODE, {
+        variables: {
+            paths, lang, uilang
+        }
+    });
+
+    useEffect(() => {
+        if (nodesInfo.data) {
+            const nodes = nodesInfo.data.jcr.nodesByPath;
+            dispatch(cePickerSetSelection(nodes.map(n => ({
```

Comment:
Just a note on this section, I saw something similar in picker configs, didn't try the behaviour but it looks like this is dependent on initialSelectedItem which I think is coming from fieldData via usePickerInputData hook in Picker2.jsx so it looks like we already have this data? https://github.com/Jahia/content-editor/blob/bc3ef5bab5ca04687dee1933db1fc498b0a2f401/src/javascript/SelectorTypes/Picker/configs/ContentPickerConfig.js#L21

Reply:
yes it's true that initialSelectedItem is coming from usePickerInputData/fieldData when using Picker2.jsx , but when using from a RichText it directly comes from the path value (which was probably the reason why the picker was taking a path and not a fieldData or a uuid). 
On the other hand, we should maybe reuse these usePickerInputData if the expected result data depends on the config .. ? Right now I'm formatting the data as in ContentPickerConfig, but MediaPickerConfig provides different values (especially for the url)